doc/BUILD.md

```diff
@@ -1,6 +1,71 @@
 Build Instructions
 ==================
 
+## 1. Platform
+* Ubuntu 18.04
+
+## 2. Build Steps
+### 2.1 ROS2
+* Install ROS2 with dependencies from the ROS2 Installation Page: https://index.ros.org/doc/ros2/Installation/
+* Install the latest distribution of ROS2 to support Navigation2 master.
+* ROS2 can be installed either from binary packages or build from source. Choose to build ROS2 from source to support Navigation2 master.
+
+### 2.2 Navigation2 Dependencies
+If ROS2 installed from source (Ignore if installed from binary packages)
+```console
+$ source ~/ros2_ws/install/local_setup.bash
+```
+Fetch, build and install navigation2 dependencies:
+```console
+$ mkdir -p ~/ros2_nav_dependencies_ws/src
+$ cd ~/ros2_nav_dependencies_ws
+$ wget https://raw.githubusercontent.com/ros-planning/navigation2/master/tools/ros2_dependencies.repos
+$ vcs import src < ros2_dependencies.repos
+$ rosdep install -y -r -q --from-paths src --ignore-src --rosdistro <latest-distro-name>
+e.g- latest-distro-name = dashing
+$ colcon build --symlink-install
+```
+
+### 2.3 Navigation2
+If ROS2 installed from source (Ignore if installed from binary packages)
+```console
+$ source ~/ros2_ws/install/local_setup.bash
+```
+Fetch, build and install navigation2 stack:
+```console
+$ source ~/ros2_nav_dependencies_ws/install/local_setup.bash
```

Comment:
`setup.bash` so you're pulling in all the parent workspace packages as well potentially built with it (turtlebot)

Reply:
That was just an anecdotal example for turtle, but people may have other workspaces they need with it, please change to `setup.bash`

Reply:
For some reason, `setup.bash` results in compilation error for me whereas `local_setup.bash` works fine. Also, prefer to include packages explicitly. Anyways, would make the change if its a strong recommendation.

Reply:
It's a strong recommendation. The typical use-case is going to be sourcing their `/opt` per-build on their `ros2_nav_ws` and if they need packages in `opt` they're not going to get them and we'll be fielding questions about "why does package XYZ not work"

Reply:
IF they do this:
```
$ source ~/ros2_ws/install/local_setup.bash
$ source ~/ros2_nav_dependencies_ws/install/local_setup.bash
```
That should be equivalent to:
```
$ source ~/ros2_nav_dependencies_ws/install/setup.bash
```
I think that's what the documentation says, right?

Reply:
You have to source them both on the same line for the best practice, and its not promised to be the same. It depends on what the environment was like when you built both other workspaces. That’s the power of setup.bash that I can compile workspaces like trees and certain ones have more workspaces linked to it for specific uses

Reply:
Looking at the `setup.bash` code, it simply sources the underlay workspaces `/install/local_setup.bash` files in order before sourcing the overlay workspace. If the underlay was the debian packages it will source it from `/opt/ros/<distro>/local_setup.bash`

Anyway, I believe that IF you know that the overlay was already built with the underlays you want, then running `setup.bash` is equivalent to running those `local_setup.bash` files then the overlay one (sorry maybe repeating myself). IF you don't know however what it was built with, sourcing the local_setup.bash is safer so you know exactly what you're getting.

However, given that the [Colcon Tutorial](https://index.ros.org//doc/ros2/Tutorials/Colcon-Tutorial/) uses setup.bash (and it's what I personally use), I'm inclined to agree with @SteveMacenski on this.

Either way I want to get this fixed and merged. We're lacking this documentation right now.

@SteveMacenski - do you mind just making the changes then we can get this in?

Reply:
@mkhansen-intel @SteveMacenski - I have made the recommended changes, kindly review.

Reply:
The new instructions look good to me. However, this PR has to be rebased before merge because the quick build section is different now.